AjuntamentdeBarcelona/decidim.barcelona
# Add follow/unfollow actions to proposal

# What and why

Users should be able to follow proposals in order to receive notifications about them. I created a new component called `FollowButton` which can be used in a polymorphic way.

The new API endpoints created are:
- `GET /api/follows`: Return a list of follows for the currrent user
- `POST /api/follows`: Create a new follow for the current user
- `DELETE /api/follows/:id`: Destroy a follow for the current user (aka unfollow).

The first two endpoints need a `following` parameter with `id` and `type` in order to scope the query. e.g.
`GET /api/follows?following[id]=1&following[type]=Proposal`
# QA

Checkout the project and check if the "Follow" button appear on `proposals#show`. After clicking on it, the "Unfollow" button should appear as well.
# GIF Tax

![](https://media.giphy.com/media/IxZgq9VVEVCBq/giphy.gif)
# TODOs
- [ ] Styles aka @josepjaume magic tricks
- [x] Unfollow test
